pkg/execute/executor_filter_test.go

```diff
@@ -0,0 +1,120 @@
+package execute
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExecutorEchoFilter_Apply(t *testing.T) {
+	var filter executorFilter = newExecutorEchoFilter()
+
+	text := "Please return this same text."
+	assert.Equal(t, text, filter.Apply(text))
+}
+
+func TestExecutorTextFilter_Apply(t *testing.T) {
+	testCases := []struct {
+		name     string
+		text     string
+		expected string
+	}{
+		{
+			name: "filter multi line text",
+			text: `NAME                                             READY   STATUS    RESTARTS   AGE
+pod/coredns-558bd4d5db-c5gwx                     1/1     Running   0          30m
+pod/coredns-558bd4d5db-j5wqt                     1/1     Running   0          30m
+pod/etcd-kind-control-plane                      1/1     Running   0          30m
+pod/kindnet-hl6zc                                1/1     Running   0          29m
+pod/kindnet-tc254                                1/1     Running   0          30m
+pod/kindnet-x79x6                                1/1     Running   0          29m
+
+NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
+daemonset.apps/kindnet      3         3         3       3            3           <none>                   30m
+daemonset.apps/kube-proxy   3         3         3       3            3           kubernetes.io/os=linux   30m`,
+			expected: `pod/etcd-kind-control-plane                      1/1     Running   0          30m
+pod/kindnet-hl6zc                                1/1     Running   0          29m
+pod/kindnet-tc254                                1/1     Running   0          30m
+pod/kindnet-x79x6                                1/1     Running   0          29m
+daemonset.apps/kindnet      3         3         3       3            3           <none>                   30m`,
+		},
+		{
+			name:     "filter single line text",
+			text:     `pod/etcd-kind-control-plane                      1/1     Running   0          30m`,
+			expected: `pod/etcd-kind-control-plane                      1/1     Running   0          30m`,
+		},
+		{
+			name:     "no match filter",
+			text:     `pod/etcd-control-plane                      1/1     Running   0          30m`,
+			expected: "",
+		},
+	}
+
+	var txFilter executorFilter = newExecutorTextFilter("kind")
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, tc.expected, txFilter.Apply(tc.text))
+		})
+	}
+}
+
+func TestExtractExecutorFilter(t *testing.T) {
+	testCases := []struct {
+		name           string
+		cmd            string
+		extractedCmd   string
+		text           string
+		filterApplied  string
+		filterDetected bool
+	}{
```

Comment:
please also handle this case:
```go
		{
			name:           "extract double quoted text filter in the middle of the command",
			cmd:            `kubectl get po  --filter="kind" --filter="test" --filter -n kube-system`,
			extractedCmd:   "kubectl get po  -n kube-system",
			text:           `etcd-control-plane                      1/1     Running   0          86m`,
			filterApplied:  "",
			filterDetected: true,
		},
```

In general:
1. What we do when user specifies the `--fliter` multiple times?
2. What we do when user specifies `--filter` (doesn't have the value)?
3. What we do when user specifies  `--filter      ` (multiple spaces)?
4. Do we want to support only such simple matching? We estimated task for 5MD, so I expected at least regex support. It was also mentioned in notes: 
   > Only grep for now / or instead of using binary, use some existing Go library